src/pudl/workspace/setup.py

```diff
@@ -1,15 +1,97 @@
 """Tools for setting up and managing PUDL workspaces."""
 import importlib
+import os
 import pathlib
 import shutil
+from pathlib import Path
+from typing import IO
 
 import yaml
+from dotenv import load_dotenv
 
 import pudl.logging_helpers
 
 logger = pudl.logging_helpers.get_logger(__name__)
 
 
+def get_defaults(
```

Comment:
I took a harder look at `get_defaults` and it looked like we were trying to use it as the one stop shop for getting the PUDL settings, and also our functionality basically strictly encompasses it (if you pass in no kwargs and have no env vars, we should get the same behavior as `get_defaults`). So I decided we should replace that instead of trying to have two similar functions next to each other.